telethon/telegram_client.py

```diff
@@ -184,6 +184,84 @@ def send_code_request(self, phone, force_sms=False):
 
         return result
 
+    def start(self, phone=None, bot_token=None, force_sms=False, code_callback=None,
+              password_callback=None):
+        """Convenience method to interactively connect and authorize.
+
+        Example usage:
+            >>> client = TelegramClient(session_id, api_id, api_hash).start(phone_number)
+            Please enter the code you received: 12345
+            Please enter your password: *******
+            (You are now logged in)
+
+        Custom `code_callback` and `password_callback` functions without arguments can be
+        provided that will be used to obtain the Telegram login code and 2FA password,
+        respectively. They will default to the `input()` function from the standard library.
+
+        Args:
+            phone (:obj:`str` | :obj:`int`):
+                The phone to which the code will be sent.
+            bot_token (:obj:`str`):
+                Bot Token obtained by @BotFather to log in as a bot.
+            force_sms (:obj:`bool`, optional):
+                Whether to force sending as SMS.
+            code_callback (:obj:`callable`, optional):
+                A callable that will be used to retrieve the Telegram login code.
+            password_callback (:obj:`callable`, optional):
+                A callable that will be used to retrieve the user's 2FA password.
+
+        Returns:
+            :obj:`TelegramClient`: This client.
+        """
+
+        if code_callback is None:
+            code_callback = lambda: input("Please enter the code you received: ")
+        else:
+            if not callable(code_callback):
+                raise ValueError("The code_callback parameter needs to be a callable function that "
+                                 "returns the code you received by Telegram.")
+
+        if password_callback is None:
+            password_callback = lambda: input("Please enter your password: ")
+        else:
+            if not callable(password_callback):
```

Comment:
Users probably know their 2FA password, and this won't let them pass the string directly.

Reply:
You're right. We had discussed that.